Enhanced reporting of TOML parse errors

wfouche opened the pull request:
```
TOML data extracted from Jython script:
     1 :requires-jython = "2.7.4"
     2 :requires-java = "21"
     3 :dependencies = [
     4 :  "io.leego:banana:2.1.0"
     5 :]
     6 :runtime-options = [
     7 :  "-Dpython.console.encoding=UTF-8"
     8 :
Unexpected end of input, expected ], a comma, or a newline (line 8, column 1)
```

wfouche committed:
Enhanced reporting of TOML parse errors

wfouche commented:
@jeff5 , the parse errors are now clearly displayed below the extract TOML text block.

Please review and merge.

jeff5 commented on JythonCli.java:
```diff
@@ -141,8 +144,20 @@ void readJBangBlock(Reader script) throws IOException {
+                for (TomlParseError err: tpr.errors()) {
+                    System.err.println(err.toString());
+                }
+                throw new IOException();
```
Since this exception is the last thing printed, it would help to have a message distinguishing this from "real" `IOException`s. Maybe "Error interpreting JBang TOML. Re-run with '--debug-cli' for details."

wfouche replied:
Fixed.

jeff5 commented:
In `TestJythonCli.java` we have tests aimed at verifying the detection of bad `jbang` blocks. They are disabled because they do not pass (and we like to see green ticks), but are just waiting for this moment.

Would you enable them please, as part of this PR, and then think about why they still fail?

I wrote them based on the discussion in #3, but they're not necessarily correct. You could add more if they do not seem to cover all the cases of interest.

jeff5 commented:
Actually, I see the tests we have are all about extracting the `jbang` block correctly (the framing part of the [inline script metadata](https://packaging.python.org/en/latest/specifications/inline-script-metadata/#inline-script-metadata) spec).

We should add one that responds to incorrect TOML extracted from correctly framed metadata, which is what you've focussed on here. We do not need to test multiple types of syntax error, as we can rely on the library for that.

wfouche committed:
IOException thrown with debug state specific message

wfouche commented:
> We should add one that responds to incorrect TOML extracted from correctly framed metadata

I'm working on creating this test.

wfouche committed:
main throws IOException

Support running individual test methods using new TestJythonCli().testMethod()

wfouche committed:
Fixed testInvalidTOML() test case

wfouche commented:
@jeff5 , really nice work you did with `TestJythonCli.java`. 

Please review the requested changes.

wfouche committed:
Allow empty TOML lines to be '#' and not required to be '# '

wfouche committed:
Use jbang-fmt to ensure consistent source file formatting

wfouche committed:
Use google-java-format to format Java source files

wfouche committed:
Removed formatting changes

wfouche committed:
Removed call to stripIndent()

wfouche commented:
@jeff5 , I made a couple of formatting changes which have all been reverted (for now).

I'm done with the updates, so over to you for a review when you have time.

jeff5 commented:
> really nice work you did with `TestJythonCli.java`.

It would be nicer if I had not messed up the indenting of the multiline strings. You may have fixed that yourself. I'll look at the latest, thanks.

jeff5 commented:
On the formatting changes that came and went, I thought some of it was an improvement, and I might not have made the mistake with indenting those strings if we used that style.

I saw `google-java-format` applied a wide right margin of (I think) 100. There is a very old style guide for the project that makes the same choice. (It not wrong for being old, just leaves a lot unanswered these days.) I find 100 is helpful for code, and just about fits my desktop, but makes comments tiring to read. Eclipse and (I discover) VSCode can make that distinction, although they differ in what they can express about a lot of fine points.

jeff5 committed:
Fix string indentation and other bugs

Multiline strings are now indented correctly. We disable tests that pass (fail to parse) for only for that reason. Typo "--debug-cli".

jeff5 commented:
I fixed a few things, mostly the result of my mistakes.

jeff5 commented:
I'd merge this, but as I've modified it a bit, and it will or ought to appear as his work, I'll leave a little time for @wfouche to agree.

It'll be a squash merge.

jeff5 reviewed (approved):
LGTM (if you accept my changes).

wfouche commented:
@jeff5, looks great to me!  Thanks.

wfouche commented:
> On the formatting changes that came and went, I thought some 
> of it was an improvement, and I might not have made the mistake with indenting those strings if we used that style.

I am planning to re-submit the formatting changes for review in another PR. The diffs were substantial and I thought it would make it more difficult to review the current PR.

jeff5 commented:
> I am planning to re-submit the formatting changes for review in another PR. The diffs were substantial and I thought it would make it more difficult to review the current PR.

That's exactly right. It is much better to make larger formatting changes a separate PR.

Would you approximate [this guide](https://wiki.python.org/jython/CodingStandards) please? Except, if your IDE will co-operate, limit the width of comments to 72 characters from the start of the comment. This is perhaps a good example of that: https://github.com/jython/jython/blob/main/core/src/test/java/org/python/core/PyUnicodeTest.java

I am experimenting also with a less strict approach to curly brackets and line wrapping for short constructs, but it takes a lot of IDE tweaking and it's not done yet.
